test/test_integration.py

```diff
@@ -0,0 +1,114 @@
+# coding: utf-8
+
+#
+# Copyright 2014-2017 Groupon, Inc.
+# Copyright 2014-2017 The Billing Project, LLC
+#
+# The Billing Project, LLC licenses this file to you under the Apache License, version 2.0
+# (the "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at:
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+"""
+    Kill Bill
+
+    Kill Bill is an open-source billing and payments platform  # noqa: E501
+
+    OpenAPI spec version: 1.0.0
+    
+    Generated by: https://github.com/swagger-api/swagger-codegen.git
+"""
+
+
+from __future__ import absolute_import
+
+import unittest
+
+import killbill
+from killbill.models.account import Account  # noqa: E501
+from killbill.models.tenant import Tenant  # noqa: E501
+from killbill.models.payment_method import PaymentMethod  # noqa: E501
+from killbill.models.subscription import Subscription  # noqa: E501
+from killbill.models.invoice_dry_run import InvoiceDryRun  # noqa: E501
+from killbill.rest import ApiException
+
+
+
+class TestIntegration(unittest.TestCase):
+    """Account unit test stubs"""
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_integration(self):
+
+        # Create tenant
+        # TODO WORKING WITH ISSUES*! [ValueError: Invalid value for `api_key`, must not be `None`]
```

Comment:
Regarding, "Invalid value for `api_key`, must not be `None`", this is invalid for the `tenant` model:

https://github.com/killbill/killbill-swagger-coden/blob/master/src/main/resources/killbill-python/model.mustache#L103

The `api_secret` is never returned. We should either get rid of `required` checks on the generation side or remove the `required=true` property in `TenantJson` (cc @sbrossie - that didn't come up with the Java client?).

Same issue with `plan_name`, etc. below.

Reply:
The meaning of the `required` field for the properties of our models -- e.g. `Tenant`--  should be understood for the creation -- `POST` . In this specific example, what this means is that the python generator should enforce to have this property set when we create the tenant but ignore it, if not set, when fetching the resource.

In a first approximation, it would be better to unplug any validation of the required fields in our models, if this makes things easier.